inst/shiny/functions/update_selectize_inputs.R

```diff
@@ -6,89 +6,52 @@
-    # Determine the selected value(s) based on the column name
-    if (column_name == "Grouping_Variables") {
-      # Find which desired grouping columns are present
-      selected_values <- intersect(desired_grouping_columns, column_names)
+    input_colname <- sub("select_", "", input_id)
+    input_info <- alternative_choices_list[[input_colname]]
+
+    # Define the expected choices for the mapping (columns and/or values)
+    potential_mappings <- c(
+      intersect(
+        c(input_colname, strsplit(input_info$mapping_alternatives, ", ")[[1]]),
+        data_colnames
+      ),
+      strsplit(input_info$Values, ", ")[[1]]
+    )
 
-      # If none are present, set to NULL
-      if (length(selected_values) == 0) {
-        selected_values <- NULL
-      }
-    } else if (column_name == "NCA_PROFILE") {
-      # Find which desired grouping columns are present
-      nca_profile_cols <- c("ATPTREF", "DOSNO", "DOSNOP")
-      selected_values <- if (any(nca_profile_cols %in% column_names)) {
-        nca_profile_cols[nca_profile_cols %in% column_names][1]
-      } else {
-        NULL
-      }
+    # Define the values to select by default (if possible)
+    selected_vals <- if (length(potential_mappings) == 0) {
+      NULL
+    } else if (input_info$is_multiple_choice) {
+      potential_mappings
     } else {
-      # For other columns, use basic logic
-      selected_values <- if (column_name %in% column_names) column_name else NULL
+      potential_mappings[[1]]
     }
-    # Update the Selectize input with the new choices and selected values
+
+    # Update the input using the defined choices and default selections
     updateSelectizeInput(
       session,
       input_id,
-      choices = c("Select Column" = "", column_names),
-      selected = selected_values
-    )
-  }
-
-  special_cases <- list(
-    "select_AVALU" = list(
-      "Dataset Columns" = column_names,
-      "Manual Units" = manual_units$concentration
-    ),
-    "select_DOSEU" = list(
-      "Dataset Columns" = column_names,
-      "Manual Units" = manual_units$dose
-    ),
-    "select_RRLTU" = list(
-      "Dataset Columns" = column_names,
-      "Manual Units" = manual_units$time
-    )
-  )
-
-  for (input_id in names(special_cases)) {
-    # Custom logic for DOSEU fallback to DOSEAU
-    selected <- {
-      # Custom logic for DOSEU fallback to DOSEAU
-      param_name <- sub("select_", "", input_id)
-      if (input_id == "select_DOSEU" && "DOSEAU" %in% column_names) {
-        "DOSEAU"
-      } else if (param_name %in% column_names) {
-        param_name
-      } else {
-        NULL
-      }
-    }
-    updateSelectizeInput(
-      session, input_id,
-      choices = c("Select Column" = "", special_cases[[input_id]]),
-      selected = selected
+      choices = list(
+        "Select Column" = "",
+        "Mapping Columns" = data_colnames,
+        "Mapping Values" =  strsplit(alternative_choices_list[[input_colname]]$Values, ", ")[[1]]
```

Comment:
nitpick: very tiny detail that I noticed
```suggestion
        "Mapping Values" = strsplit(alternative_choices_list[[input_colname]]$Values, ", ")[[1]]
```